akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -982,6 +990,32 @@ abstract class ShardCoordinator(
     rebalanceInProgress = rebalanceInProgress.updated(shard, rebalanceInProgress(shard) + from)
   }
 
+  private def informAboutCurrentShards(ref: ActorRef): Unit = {
+    // hardcoded to a safe low value rather than configurable, for now
+    val batchSize = 500
+    if (state.shards.isEmpty) {
+      // No shards - NOP
+    } else if (state.shards.size <= batchSize) {
+      log.debug("{}: Informing [{}] about all [{}] shards", typeName, ref, state.shards.size)
+      ref ! ShardHomes(state.regions)
```

Comment:
How is the interplay with rebalance here? In the rebalance process shard locations are first deallocated to clear the caches and we should not send out ShardHomes that includes any rebalance in progress. Don't remember exactly how the state and rebalanceInProgress are updated.

Reply:
Good catch, the shards stay in state while rebalancing and there is separate logic to delay a GetShardHome request if the shard is rebalancing.

We need to filter out all shards that are rebalancing in these messages.